test-specifications/token-service/tokenCreateTransaction.md

```diff
@@ -0,0 +1,1150 @@
+| 7       | Creates an NFT with a nonzero decimal amount                                   | name="testname", symbol="testsymbol", decimals=3, treasuryAccount=<OPERATOR_ACCOUNT_ID>, tokenType="nft" | The token creation fails with an INVALID_TOKEN_DECIMALS response code from the network. | N                 |
+
+#### JSON Request Example
+
+```json
+{
+  "jsonrpc": "2.0",
+  "id": 8895,
+  "method": "createToken",
+  "params": {
+    "name": "testname",
+    "symbol": "testsymbol",
+    "decimals": 3,
+    "treasuryAccount": "0.0.2"
+  }
+}
+```
+
+#### JSON Response Example
+
+```json
+{
+  "jsonrpc": "2.0",
+  "id": 8895,
+  "result": {
+    "tokenId": "0.0.68721",
+    "status": "SUCCESS"
+  }
+}
+```
+
+### **Initial Supply:**
+
+- The number of tokens to be created.
+
+| Test no | Name                                                                            | Input                                                                                                          | Expected response                                                                                  | Implemented (Y/N) |
+|---------|---------------------------------------------------------------------------------|----------------------------------------------------------------------------------------------------------------|----------------------------------------------------------------------------------------------------|-------------------|
+| 1       | Creates a fungible token with a valid initial supply                            | name="testname", symbol="testsymbol", initialSupply=1000000, treasuryAccount=<OPERATOR_ACCOUNT_ID>             | The token creation succeeds and 1,000,000 tokens are held by the treasury account.                 | N                 |
+| 2       | Creates a fungible token with the minimum initial supply                        | name="testname", symbol="testsymbol", initialSupply=0, treasuryAccount=<OPERATOR_ACCOUNT_ID>                   | The token creation succeeds and 0 tokens are held by the treasury account.                         | N                 |
+| 3       | Creates a fungible token with an initial supply below the minimum amount        | name="testname", symbol="testsymbol", initialSupply=-1, treasuryAccount=<OPERATOR_ACCOUNT_ID>                  | The token creation fails with an INVALID_TOKEN_INITIAL_SUPPLY response code from the network.      | N                 |
+| 4       | Creates a fungible token with the maximum initial supply                        | name="testname", symbol="testsymbol", initialSupply=9223372036854775807, treasuryAccount=<OPERATOR_ACCOUNT_ID> | The token creation succeeds and 9,223,372,036,854,775,807 tokens are held by the treasury account. | N                 |
```

Comment:
Note, the result amounts here are whole tokens only if `decimals == 0`.  The actual supply is defined in terms of the smallest unit of the token (10<sup>`-decimals`</sup> whole tokens).

Reply:
Do you think it'd be worth to add a test here which has the max `initialSupply` with decimals?

Reply:
I would at least consider having a few supply tests duplicated with a non-zero decimals value.

Reply:
Added a few more tests here